iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImpl.java

```diff
@@ -95,7 +97,18 @@ public Kind getKind() {
   }
 
   @Override
-  public Identifier name() {
+  public StringLiteral name() {
+    return resourceProperties().stream()
+      .filter(prop -> TextUtils.isValue(prop.key(), "name").isTrue())
+      .filter(prop -> prop.value().is(Kind.STRING_COMPLETE))
+      .findFirst()
+      .map(prop -> ((StringComplete) prop.value()).content())
+      .orElse(null);
+  }
```

Comment:
We should add a `@CheckForNull` on this method, as now we can expect it to sometimes return a null value.